### DISCUSSION TITLE
[2024-02-14] This Bi-Weekly in Dora-rs

### DISCUSSION THREAD

**Discussion by haixuanTao** [dora-rs/dora#426] [Status Update]
Happy Chinese New Year :dragon:

**Comment by haixuanTao** [comment]
# Community

## FOSDEM

- Great reception with 400 people attending our presentation. :)
- Open invitation to participate next year.
- Interest for both individuals and small companies :)
- Link to the video:
   - Short Demo Version: https://www.youtube.com/watch?v=NvvTEP8Jak8 
   - Long Talk Version: https://video.fosdem.org/2024/ud2120/fosdem-2024-3225-dora-rs-simplifying-robotics-stack-for-next-gen-robots.mp4

## RustNL

- 30min: dora-rs main presentation
- 15min: system engineering with LLM in mind
    - From logging the logs, data, metrics, tracing standpoint
    - Using them for training, testing, benchmarking
    - to adding new capabilities with hot-reloading features

## AI_dev

AI_dev is a nexus for developers delving into the intricate realm of open source generative AI and machine learning. At the heart of this event is the belief that open source is the engine of innovation in AI. By uniting the brightest developers from around the world, we aim to ignite discussions, foster collaborations, and shape the trajectory of open source AI.

- https://events.linuxfoundation.org/ai-dev-europe/

## ICRA 2024 VLLM Workshop

-> https://vlmnm-workshop.github.io/

- Could it be good to formulate in an academical way our research in adding new capabilities using llm for dora.

- Going to work also on the VLM (Vision-Language Model) part to be able to do "Image to Code" with CentraleSupelec.


## WIP: Physical Autonomous Driving Competition

**Reply by haixuanTao** [reply]
ACCV: https://accv2024.org/ for the autonomous driving part of dora-rs. 
Submission deadline:
April 26, 2024: Workshop proposal deadline
June 28, 2024: Tutorial proposal deadline
July 2, 2024: Regular paper submission deadline

**Reply by haixuanTao** [reply]
https://arxiv.org/pdf/2402.08268.pdf

**Reply by haixuanTao** [reply]
https://hycha24.sciencesconf.org/

**Comment by phil-opp** [comment]
## ROS2 bridge support for C++ nodes

The implementation is almost ready to be merged. See https://github.com/dora-rs/dora/pull/425 for details.

Usage will look like this:

```c++
    auto dora_node = init_dora_node();
    auto merged_events = dora_events_into_combined(std::move(dora_node.events));

    auto qos = qos_default();
    qos.durability = Ros2Durability::Volatile;
    qos.liveliness = Ros2Liveliness::Automatic;
    qos.reliable = true;
    qos.max_blocking_time = 0.1;

    auto ros2_context = init_ros2_context();
    auto node = ros2_context->new_node("/ros2_demo", "turtle_teleop");
    auto vel_topic = node->create_topic_geometry_msgs_Twist("/turtle1", "cmd_vel", qos);
    auto vel_publisher = node->create_publisher(vel_topic, qos);
    auto pose_topic = node->create_topic_turtlesim_Pose("/turtle1", "pose", qos);
    auto pose_subscription = node->create_subscription(pose_topic, qos, merged_events);

    auto event = merged_events.next();

    if (event.is_dora())
    {
        auto dora_event = downcast_dora(std::move(event)); 
        ...
        geometry_msgs::Twist twist = {
                    .linear = {.x = 1, .y = 0, .z = 0},
                    .angular = {.x = 0, .y = 0, .z = 1}
        };
        vel_publisher->publish(twist);
    }
    else if (pose_subscription->matches(event))
    {
        auto pose = pose_subscription->downcast(std::move(event));
        std::cout << "Received pose x:" << pose.x << ", y:" << pose.y << std::endl;
    }
```

I'm currently working on documenting the build and usage instructions in a new [`apis/c++/node/README.md`](https://github.com/dora-rs/dora/blob/2d6b5b98079690189ccd9816d59af2d821c75d75/apis/c%2B%2B/node/README.md) file.

**Reply by haixuanTao** [reply]
Google Rust C++ Interoperability

**Reply by haixuanTao** [reply]
Create a github or dora blogpost and translate community effort.

**Reply by haixuanTao** [reply]
https://github.com/dora-rs/dora/wiki
